csharp/src/Ice/UriParser.cs

```diff
@@ -333,6 +368,18 @@ internal static void RegisterCommon()
                 }
             }
             return (uri, altEndpoint, proxyOptions);
+
+            void CheckProxyOption(string name, bool alreadySet)
```

Comment:
We can consider adding `pureEndpoints` and `uriString` and make this static

Reply:
I don't see the benefit of switching to a static local function here.